Sources/FluentKit/Properties/OptionalParent.swift

```diff
@@ -151,25 +151,25 @@ extension OptionalParentProperty: EagerLoadable {
 }
 
 private struct OptionalParentEagerLoader<From, To>: EagerLoader
-    where From: Model, To: Model
+    where From: FluentKit.Model, To: FluentKit.Model
 {
-    let relationKey: KeyPath<From, From.OptionalParent<To>>
+    let relationKey: KeyPath<From, OptionalParentProperty<From, To>>
 
     func run(models: [From], on database: Database) -> EventLoopFuture<Void> {
-        let ids = models.compactMap {
-            $0[keyPath: self.relationKey].id
-        }
-        
-        return To.query(on: database)
-            .filter(\._$id ~~ Set(ids))
-            .all()
-            .map
-        {
-            for model in models {
-                model[keyPath: self.relationKey].value = .some($0.filter {
-                    $0.id == model[keyPath: self.relationKey].id
-                }.first)
+        var sets = Dictionary(grouping: models, by: { $0[keyPath: self.relationKey].id })
+        let nilParentModels = sets.removeValue(forKey: nil) ?? []
+
+        return To.query(on: database).filter(\._$id ~~ Set(sets.keys.compactMap { $0 })).all().flatMapThrowing {
+            let parents = Dictionary(uniqueKeysWithValues: $0.map { ($0.id!, $0) })
+
+            for (parentId, models) in sets {
+                guard let parent = parents[parentId!] else {
+                    database.logger.debug("No parent '\(To.self)' with id '\(parentId!)' was found in eager-load results.")
```

Comment:
Parent ID should be metadata instead if embedded in the String

Reply:
Honestly, I only even added this log line for @jdmcd's benefit, and later mooted the point by making the call to slightly break API by throwing a useful (**GASP!!!**) error; it probably can just be removed.